annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,283 @@
+package edu.isi.vista.annotationutils
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import edu.isi.nlp.parameters.serifstyle.SerifStyleParameterFileLoader
+import kotlinx.html.*
+import kotlinx.html.stream.appendHTML
+import org.nield.kotlinstatistics.countBy
+import java.io.File
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * This program is designed to collect the annotation
+ * statistics from the exported JSON, and they will get pushed along with
+ * the updated annotation files.
+ */
+class ExtractAnnotationStats {
+    companion object {
+        fun main(argv: Array<String>) {
+            if (argv.size != 1) {
+                throw RuntimeException("Expected a single argument: a parameter file")
+            }
+            val paramsLoader = SerifStyleParameterFileLoader.Builder()
+                    .interpolateEnvironmentalVariables(true).build()
+            val params = paramsLoader.load(File(argv[0]))
+            extractStats(params)
+        }
+        fun extractStats(params: edu.isi.nlp.parameters.Parameters) {
+            val exportAnnotationRoot = params.getExistingDirectory("exportedAnnotationRoot")
+            val statisticsDirectory = params.getExistingDirectory("statisticsDirectory")
+            val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+            TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+            val formatter = SimpleDateFormat("yyyy-MM-dd")
+            val thisDate = formatter.format(currentDate.getTime())
+
+            val documentsWithEventAnnotations: MutableList<DocumentAnnotation> = arrayListOf()
+            collectStats(exportAnnotationRoot, documentsWithEventAnnotations)
+
+            // From documentsWithEventAnnotations, get TOTAL, TOTAL PER USER and TOTAL PER EVENT TYPE
+            val totalAnnotations = documentsWithEventAnnotations.size
+            val annotationsByUser = documentsWithEventAnnotations.countBy { it.user }
+            val annotationsByEventType = documentsWithEventAnnotations.countBy { it.eventType }
+            // For better organization
+            val sortedUsers = annotationsByUser.toSortedMap()
+            val sortedEventTypes = annotationsByEventType.toSortedMap()
+            val newAnnotationStats = AnnotationStats(totalAnnotations, sortedUsers, sortedEventTypes)
+
+            // Load the annotation statistics from the last run
+            val previousStatsReport: StatsReport? = locatePreviousStats(statisticsDirectory)
+            var previousAnnotationStats: AnnotationStats? = null
+            var annotationDiffs: AnnotationStats? = null
+            if (previousStatsReport == null) {
```

Comment:
nit: it's somewhat more readable if you put the "expected/more interesting" branch of the "if" first and the warning/error condition last.